src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -2207,7 +2207,7 @@ public override BoundNode VisitLocalDeclaration(BoundLocalDeclaration node)
 
             if (node.InitializerOpt != null)
             {
-                VisitRvalue(node.InitializerOpt); // analyze the expression
+                base.VisitLocalDeclaration(node);
```

Comment:
If you removed the `if` condition entirely and simply visited using `base.Visit`, would expression visitors take care of calling `Assign` on nested variables?